File: tests/fixtures/language_adapter_symbols/sample.lua
-- HAND-DERIVED: standard Lua syntax written from language knowledge, not from any adapter regex.
local Warehouse = {}
Warehouse.__index = Warehouse

function Warehouse.new()
    local self = setmetatable({}, Warehouse)
    self.items = {}
    return self
end

function Warehouse:addItem(name, quantity)
    table.insert(self.items, { name = name, quantity = quantity })
end

return Warehouse

File: tests/fixtures/language_adapter_symbols/sample.ex
# HAND-DERIVED: standard Elixir syntax written from language knowledge, not from any adapter regex.
defmodule Inventory.Warehouse do
  defstruct items: []

  def add_item(%__MODULE__{items: items} = warehouse, name, quantity) do
    %{warehouse | items: [{name, quantity} | items]}
  end

  def total_quantity(%__MODULE__{items: items}) do
    Enum.reduce(items, 0, fn {_name, qty}, acc -> acc + qty end)
  end
end
